src/main/kotlin/LocalEvaluationClient.kt

```diff
@@ -67,15 +102,115 @@ class LocalEvaluationClient internal constructor(
 
     @JvmOverloads
     fun evaluateV2(user: ExperimentUser, flagKeys: Set<String> = setOf()): Map<String, Variant> {
-        val flagConfigs = flagConfigService.getFlagConfigs().toMap()
+        val flagConfigs = flagConfigStorage.getFlagConfigs()
         val sortedFlagConfigs = topologicalSort(flagConfigs, flagKeys)
         if (sortedFlagConfigs.isEmpty()) {
             return mapOf()
         }
-        val evaluationResults = evaluation.evaluate(user.toEvaluationContext(), sortedFlagConfigs)
-        Logger.d("evaluate - user=$user, result=$evaluationResults")
+        val enrichedUser = enrichUser(user, sortedFlagConfigs)
+        val evaluationResults = wrapMetrics(
+            metric = metrics::onEvaluation,
+            failure = metrics::onEvaluationFailure,
+        ) {
+            evaluation.evaluate(enrichedUser.toEvaluationContext(), sortedFlagConfigs)
+        }
         val variants = evaluationResults.toVariants()
         assignmentService?.track(Assignment(user, variants))
         return variants
     }
+
+    private fun enrichUser(user: ExperimentUser, flagConfigs: List<EvaluationFlag>): ExperimentUser {
+        val groupedCohortIds = flagConfigs.getGroupedCohortIds()
+        if (cohortStorage == null) {
+            if (groupedCohortIds.isNotEmpty()) {
+                val flagKeys = flagConfigs.mapNotNull { flag ->
+                    val cohortIds = flag.getAllCohortIds()
+                    if (cohortIds.isEmpty()) {
+                        null
+                    } else {
+                        flag.key
+                    }
+                }
+                Logger.e("Local evaluation flags $flagKeys target cohorts but cohort targeting is not configured.")
```

Comment:
during evaluation, should log warning for a flag if cohort targeting is configured, but cohort not in storage (due to e.g. download failure?)